src/Tasks/Microsoft.NET.Build.Tasks/build/Microsoft.NET.GenerateAssemblyInfo.targets

```diff
@@ -84,7 +81,35 @@ Copyright (c) .NET Foundation. All rights reserved.
         <_Parameter1>$(NeutralLanguage)</_Parameter1>
       </AssemblyAttribute>
     </ItemGroup>
+  </Target>
+
+  <!-- 
+    To allow version changes to be respected on incremental builds (e.g. through CLI parameters),
+    create a hash of all assembly attributes so that the lock file name will change with the calculated
+    assembly attribute values and msbuild will then execute CoreGenerateAssembly to generate a new file.
+  -->
+  <Target Name="_CreateGeneratedAssemblyInfoInputsCacheFile"
+          DependsOnTargets="_CalculateAssemblyAttributes"
+          BeforeTargets="CoreGenerateAssemblyInfo">
```

Comment:
Nothing accidental, but it removes flexibility to invoke the individual targets via something other than GenerateAssemblyInfo and still have its dependencies satisfied. You've convinced me that the explicit approach is nicer.

Thinking further, given

1. Error handling subtleties of Microsoft/msbuild#2133
2. That we need CoreGenerateAssemblyInfo overrides to re-declare the inputs and outputs. (I had put them on outer GenerateAssemblyInfo and that didn't work.
3. That one of the inputs is the generated cache file

It makes sense for CoreGenerateAssemblyInfo to explicitly state that it depends on creating the cache file.

So here's my proposal:

``` xml
<Target Name="GenerateAssemblyInfo" 
        DependsOnTargets="PrepareForBuild;CoreGenerateAssemblyInfo"
        BeforeTargets="CoreCompile" ... />

<Target Name="GetAssemblyAttributes"
        DependsOnTargets="GetAssemblyVersion" .. />

<Target Name="CreateGeneratedAssemblyInfoInputsCacheFile"
        DependsOnTargets="GetAssemblyAttributes" />

<Target Name="CoreGenerateAssemblyInfo" 
       DependsOnTargets="CreateGeneratedAssemblyInfoInputsCacheFile"
       Inputs="$(GeneratedAssemblyInfoCacheFile)"
       Outputs="$(GeneratedAssemblyInfoFile)" ... >